File: Week_02/inorderTraversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func inorderTraversal(root *TreeNode) []int {
    res:=[]int{}
    var traver func(root *TreeNode)
    traver=func(root *TreeNode){ 
        if root!=nil{
            traver(root.Left)
            res=append(res,root.Val)
            traver(root.Right)
        }
    }
    traver(root)
    return res
}

File: Week_02/inorderTraversal.py
class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        res=[]
        
        def traver(root):
            if root:
                traver(root.left)
                res.append(root.val)
                traver(root.right)
        traver(root)
        
        return res
